### PR TITLE
[FIX] account_invoice

### PR DESCRIPTION
**Pull request by nim-odoo**

### CODE REVIEW

File: addons/account/account_invoice.py [hunk 1]
```diff
@@ -1359,8 +1360,9 @@ def product_id_change(self, product, uom_id, qty=0, name='', type='out_invoice',
             if company.currency_id != currency:
                 if type in ('in_invoice', 'in_refund'):
                     values['price_unit'] = product.standard_price
-                values['price_unit'] = values['price_unit'] * currency.rate
-
+                ipdb.set_trace()  ######### Break Point ###########
```
Comment (rvalyi) [inline addons/account/account_invoice.py]: I doubt you really want that line to be merged

File: openerp/addons/base/res/res_currency.py [hunk 2]
```diff
@@ -45,6 +46,8 @@ def _get_current_rate(self, cr, uid, ids, raise_on_no_rate=True, context=None):
         res = {}
 
         date = context.get('date') or time.strftime('%Y-%m-%d')
+        ipdb.set_trace()  ######### Break Point ###########
```
Comment (rvalyi) [inline openerp/addons/base/res/res_currency.py]: same here

**Comment by beledouxdenis** [comment]
As the `rate` field use the method `_get_current_rate`,
I believe you can do this
`currency.with_context(dict(context or {}, date=date_invoice)).rate`
instead of
`currency.with_context(dict(context or {}, date=date_invoice))._get_current_rate()[currency_id]`
Which is prettier and provide a shorter diff. Can you give it a try ?

Besides that, :+1:

**Comment by nim-odoo** [comment]
It's working with your suggestion ;) OK to merge?

**Comment by beledouxdenis** [comment]
If you made sure you didn't miss a place, yes you can merge :).

**Comment by nim-odoo** [comment]
Thanks for making me doubt :-P

**Pull request merged**